Chat | add logging in front-end app

macbre opened the pull request:
Add `?log_level=13` to the URL to actually see the logs in the browser console.

See https://github.com/Wikia/chat/pull/50

Now I'm getting these on my devbox:

```
chat_js2:519 chat:  Connecting to chat server: http://dev-macbre:8080 ...
chat_js2:519 chat:  Authenticating using MediaWiki: ...
chat_js2:519 chat:  Connected to Chat server at http://dev-macbre:8080
chat_js2:519 chat:  socket.onerror: Error: server error - parser error
chat_js2:519 chat:  Sending "initquery" command...
chat_js2:519 chat:  Reconnecting...
chat_js2:519 chat:  Connected to Chat server at http://dev-macbre:8080
chat_js2:519 chat:  socket.onerror: Error: server error - parser error
chat_js2:519 chat:  Sending "initquery" command...
chat_js2:519 chat:  Reconnecting...
...
```

And remove formatting of `editCount` as it was breaking socket.io package parsing in Chat browser app causing the connection drop and re-connection (and so on...):

```
...\"editCount\":\"12Â 584\"...
```

`Language::formatNum` uses UTF separator for digit groups (at least in Polish).

macbre committed:
Chat | add logging in front-end app

Add ?log_level=13 to the URL to actually see the logs in the browser console.

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L105](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L105): Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L187](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L187): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L227](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L227): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L242](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L242): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L267](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L267): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L284](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L284): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L105](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L105): Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L187](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L187): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L227](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L227): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L242](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L242): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L267](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L267): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L284](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L284): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
More logging

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L105](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L105): Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L151](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L151): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L187](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L187): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L227](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L227): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L242](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L242): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L267](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L267): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L284](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L284): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/9a7cebf84536847f7c07b5b9acc6b6daa19f252e/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L159](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L159): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L191](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L191): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L246](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L246): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L271](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L271): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L288](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L291](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L291): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L347](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L347): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented on extensions/wikia/Chat2/js/controllers/controllers.js:
```diff
@@ -91,6 +103,9 @@ var NodeChatSocketWrapper = $.createClass(Observable, {
+		this.log('Message received: ' + message.event);
+		this.log(message);
+
 		switch (message.event) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)

macbre committed:
socket.onerror - log details

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L159](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L159): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L191](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L191): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L246](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L246): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L271](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L271): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L288](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L291](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L291): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L347](https://github.com/Wikia/app/blob/782edcdb07fd5ac7c4358a0d07371bb5813afb82/extensions/wikia/Chat2/js/controllers/controllers.js#L347): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L159](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L159): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L191](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L191): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L246](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L246): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L271](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L271): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L288](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L291](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L291): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L347](https://github.com/Wikia/app/blob/b9e43d884294abaa2824b8e0d0787e9ce3be36d1/extensions/wikia/Chat2/js/controllers/controllers.js#L347): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
NodeChatSocketWrapper: indentiation fix

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 47 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L159](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L159): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L191](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L191): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L246](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L246): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L271](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L271): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L288](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L291](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L291): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L347](https://github.com/Wikia/app/blob/7b8792c11c786dc0e3d2cb1e857bf6a783cc5fb2/extensions/wikia/Chat2/js/controllers/controllers.js#L347): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

mszabo-wikia reviewed (changes requested):
Should we log every event in chat for every user?

macbre commented:
Ignore Jenkins fails - it tried to fetch ` > git rev-parse IRIS-4535^{commit} # timeout=10`

macbre committed:
Group logs

WikiaTech commented on extensions/wikia/Chat2/js/controllers/controllers.js:
```diff
@@ -635,7 +650,7 @@ var NodeChatController = $.createClass(NodeRoomController, {
 	showRoom: function (roomId) {
-		$().log(roomId);
+		$().log(roomId, 'chat:showRoom');
 		if (this.activeRoom == roomId) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

WikiaTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

WikiaTech commented:
SonarQube analysis reported 66 issues
* ![MAJOR][MAJOR] 55 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L155](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L155): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L159](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L159): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L191](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L191): Replace "!=" with "!==". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Remove the unused function parameter "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L246](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L246): Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L271](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L271): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L288](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L291](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L291): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [controllers.js#L347](https://github.com/Wikia/app/blob/894c5214fdc5f09bcf8355e843acecc2a6f60ce5/extensions/wikia/Chat2/js/controllers/controllers.js#L347): Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
ChatAjax: do not emit UTF characters in editCount field

Language::formatNum returns UTF characters for Polish that socket.io can't decode in Chat browser app.

WikiaTech commented:
SonarQube analysis reported 82 issues
* ![MAJOR][MAJOR] 69 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 11 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L29): Remove this unused "$wgContLang" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L119](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L119): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

WikiaTech commented:
SonarQube analysis reported 82 issues
* ![MAJOR][MAJOR] 69 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 11 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L29): Remove this unused "$wgContLang" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L119](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L119): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/283666e5dd89f424570f8e1e1e0f2992e95902a1/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

macbre committed:
Merge branch 'dev' into chat-logging

WikiaTech commented:
SonarQube analysis reported 83 issues
* ![MAJOR][MAJOR] 70 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 11 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatAjax.class.php#L28](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L29): Remove this unused "$wgContLang" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L29](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L119](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L119): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L142](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L142): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L173](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L173): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatAjax.class.php#L231](https://github.com/Wikia/app/blob/60cc0d6280235c54dbfbf867d935190166f84492/extensions/wikia/Chat2/ChatAjax.class.php#L231): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
